tests/test_nonnc.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+  Copyright 2024 Dr. Blatt - HPC-Simulation-Software & Services
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+
+#define BOOST_TEST_MODULE NONNCTest
+#define BOOST_TEST_NO_MAIN
+
+#include <opm/input/eclipse/Deck/Deck.hpp>
+#include <opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <opm/input/eclipse/Parser/Parser.hpp>
+#include <opm/input/eclipse/EclipseState/EclipseState.hpp>
+
+#include <dune/grid/common/mcmgmapper.hh>
+
+#include <opm/grid/CpGrid.hpp>
+
+#include <ebos/ecltransmissibility.cc>
+// Include ecltransmissibility_impl.hh as well, because we need the isId and isIdReverse functions that manage the keys of transmissibility map
+#include <ebos/ecltransmissibility_impl.hh>
```

Comment:
there's no reason to include the impl file here. just include ecltransmissibility.hh. The code is in the library.

Reply:
I need the functions isId and isIdReverse that are defined as free functions in ecltransmissibility_impl.hh though... of course it is not nice to include the _impl.hh file - I'll glady accept any suggestions!

Reply:
i see. then de-anonymize them, ie. put them in the Opm::details namespace and put prototypes in the .hh.